src/main/java/org/wso2/carbon/securevault/hashicorp/config/HashiCorpVaultConfigLoader.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.securevault.hashicorp.config;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.securevault.hashicorp.exception.HashiCorpVaultException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.CONFIG_FILE_PATH;
+
+/**
+ * Configuration Loader for Vault Configurations.
+ */
+public class HashiCorpVaultConfigLoader {
+
+    private static final Log LOG = LogFactory.getLog(HashiCorpVaultConfigLoader.class);
+
+    private static HashiCorpVaultConfigLoader instance = null;
+
+    private Properties properties;
+
+    private HashiCorpVaultConfigLoader() {
+
+        try {
+            loadConfigurations();
```

Comment:
Having this load mechanism at class construction causes the system startup to slow down, may be by few milliseconds. But we need to avoid that as few of these soon aggregate to few seconds startup delay. The correct way is to do the load at getInstance() method itself.

Reply:
Ack!